bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -421,11 +421,24 @@ private static byte[] parseHexString(final String s) {
 
         byte[] b = new byte[OBJECT_ID_LENGTH];
         for (int i = 0; i < b.length; i++) {
-            b[i] = (byte) Integer.parseInt(s.substring(i * 2, i * 2 + 2), 16);
+            char c1 = s.charAt(i * 2);
+            char c2 = s.charAt(i * 2 + 1);
+            b[i] = hexCharToByte(c1) * 16 + hexCharToByte(c2)
```

Comment:
This doesn't compile due to a missing semicolon.

Reply:
Sorry about that, was (sloppily) converting the code from Kotlin. Have resolved this and other style issues and verified passing `bson:check`